packages/expo-image-picker/src/ImagePicker.types.ts

```diff
@@ -38,19 +38,31 @@ export type ImageInfo = {
 
 export type ImagePickerResult = { cancelled: true } | ({ cancelled: false } & ImageInfo);
 
+export type ImagePickerMultipleResult = { cancelled: true } | ({ cancelled: false; selected: ImageInfo[] });
+
+
 export type ImagePickerOptions = {
   allowsEditing?: boolean;
   aspect?: [number, number];
   quality?: number;
-  allowsMultipleSelection?: boolean;
   mediaTypes?: MediaTypeOptions;
   exif?: boolean;
   base64?: boolean;
   videoExportPreset?: VideoExportPreset;
+   allowsMultipleSelection?: boolean;
 };
 
 export type OpenFileBrowserOptions = {
   mediaTypes: MediaTypeOptions;
   capture?: boolean;
   allowsMultipleSelection: boolean;
 };
+
+
+export type ExpandImagePickerOptions =
+  | (ImagePickerOptions & { allowsMultipleSelection: true })
+  | ImagePickerOptions;
+
+export type ExpandImagePickerResult<T> = T extends ImagePickerOptions & { allowsMultipleSelection: true }
+  ? ImagePickerMultipleResult
+  : ImagePickerResult
```

Comment:
Getting rid of `ExpandImagePickerOptions` would make us write sth like this one:
```suggestion
export type ExpandImagePickerResult<T extends ImagePickerOptions> = T extends { allowsMultipleSelection: true }
  ? ImagePickerMultipleResult
  : ImagePickerResult

```
Here's more contextual example: [typescript playground](https://www.staging-typescript.org/play?#code/KYDwDg9gTgLgBDAnmYcCSBbAhgc2GgOwDMI4BeOAbwF8BuAKHtElgWVU12AAUBLAYwDWwKAHkwMXhAIBnclThYANkogB3GQFkArkslglwAMrBD-SdID8ALjgAjCBENYCtOHUbNo8JCnTY8PiERACVgGV14Cko4fhd+U0MAE1sYKG1UajgAHzgAChi4ggSVYBS4ImUZTLgAMn8uQhIASgYvVl8OAJ4BYSgdPV4DYDCIvXlC+MSy1PSa3ILYqdLyyqVqt2qzGBmGvCaIAG0AXWpWz3BvNj8AUXAXJM5A3tDwyIAeABU4UB2CJLkTx6wTEEiksgAfPJvr9gP85DFlKoNAN9IYTNtwbMMu56HA4JY9sC+qihoZRpE8XBbECgn0KXpGExLqwsDJEMUKtpihYCHAlFhufwABZAgAyvDsUCwUEQAEF2cUvj8QH8AUS6SJxLyZBC8hAwdIZLZPs1bNwoBAMLxqu87mAHrSXlAGTAvhCoZQqQB6b1wDDAGDCiBJewhxD0aiMX1wT7heDVczguAXFjwIhC3mKGTVWCfdjuvIAfRNzSoUa9+LZuZg+ZQ7ydINdeqwaiwvHgAqFou6EqlMvliv4BTO5yrOZEtYLjfpbz0LbbHf5guKPa4felsoVHOHiJU6i0kTJxlMwCT0ls3KSwCIvAIZXczTH2ZrdeADe6mpdc5gC-bnZXEVxUlTdBx3RYkQPUlhgxM9eVsNZqkfZ9q0nN8Py4L9oPJH8-yXLtV2A-styHCD9xRI8YNPc8CGxTInwYaggA)